### PR TITLE
[BH-1940] New progress time widget

### PR DESCRIPTION
**Pull request by dawidwojtas-mudita**
This widget allows to display of minutes and also, seconds when 1 minute is left to the end of the timer.

Make sure that this PR:
- [x] Complies with our guidelines for contributions
- [ ] Has unit tests if possible
- [ ] Has documentation updated
- [ ] Has changelog entry added

### CODE REVIEW

File: module-apps/apps-common/widgets/AbstractProgressTime.hpp [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+#pragma once
+
+#include <cstdint>
+
+namespace gui
+{
+    namespace progress
+    {
+        using namespace std::chrono_literals;
+        constexpr auto secondsInMinute = std::chrono::seconds(1min).count();
```
Comment (Lefucjusz) [inline module-apps/apps-common/widgets/AbstractProgressTime.hpp:13]: You can use `utils::time::secondsInMinute` constant from `module_utils\time\time\time_constants.hpp` :wink: If you prefer to leave it as it is, please add `inline` to the declaration.

**Review by Lefucjusz — APPROVED** [review]

File: module-apps/apps-common/widgets/AbstractProgressTime.hpp (hunk 1 above)
Comment (dawidwojtas-mudita) [reply module-apps/apps-common/widgets/AbstractProgressTime.hpp]: Thanks! I decided to use this one from `utils::time::secondsInMinute` so it's done ;)

File: module-apps/apps-common/widgets/ProgressTimerWithBarGraphAndCounter.hpp [hunk 2]
```diff
@@ -20,9 +20,9 @@ namespace app
 {
     class ProgressTimerWithBarGraphAndCounter : public ProgressTimer
     {
-        gui::Text *text                  = nullptr;
-        gui::Progress *progress          = nullptr;
-        gui::TimeFixedWidget *timeWidget = nullptr;
+        gui::Text *text                       = nullptr;
+        gui::Progress *progress               = nullptr;
+        gui::AbstractProgressTime *timeWidget = nullptr;
```
Comment (Maciej-Mudita) [inline module-apps/apps-common/widgets/ProgressTimerWithBarGraphAndCounter.hpp:25]: please consider:
```suggestion
        gui::Text *text{nullptr};
        gui::Progress *progress{nullptr};
        gui::AbstractProgressTime *timeWidget{nullptr};
```

File: module-apps/apps-common/widgets/TimeMinuteSecondWidget.hpp [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+                               DisplayType type = DisplayType::OnlyMinutes);
+
+        void updateTime(std::uint32_t sec) override;
+        void buildInterface(std::uint32_t w, std::uint32_t h);
+        void setText(std::uint32_t value);
+
+      private:
+        DisplayType displayType;
+
+        static constexpr auto maxDigits{3U};
```
Comment (Maciej-Mudita) [inline module-apps/apps-common/widgets/TimeMinuteSecondWidget.hpp:35]: can this be moved to an anonymous space in the cpp file?

**Review by Maciej-Mudita — APPROVED** [review]

File: module-apps/apps-common/widgets/ProgressTimerWithBarGraphAndCounter.hpp (hunk 2 above)
Comment (dawidwojtas-mudita) [reply module-apps/apps-common/widgets/ProgressTimerWithBarGraphAndCounter.hpp]: Thanks, done ;)

File: products/BellHybrid/apps/application-bell-focus-timer/windows/FocusTimerWindow.hpp [hunk 4]
```diff
@@ -37,7 +37,7 @@ namespace app::focus
         std::unique_ptr<FocusTimerContract::Presenter> presenter;
         gui::VBox *mainVBox                   = nullptr;
         gui::ArcProgressBar *progress         = nullptr;
-        gui::TimeFixedWidget *timer           = nullptr;
+        gui::TimeMinuteSecondWidget *timer    = nullptr;
         gui::TextFixedSize *bottomDescription = nullptr;
         gui::Icon *iconPause                  = nullptr;
         gui::Icon *iconRing                   = nullptr;
```
[the remark this answers is not in the document]
Comment (dawidwojtas-mudita) [reply products/BellHybrid/apps/application-bell-focus-timer/windows/FocusTimerWindow.hpp]: Done

**Pull request merged**